### PR TITLE
Add HTTP status code to `HttpResponseError`

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
Part of [ML-5598](https://jira.iguazeng.com/browse/ML-5598).

### CODE REVIEW

File: v3io/dataplane/response.py [hunk 1]
```diff
@@ -22,7 +22,9 @@
 class HttpResponseError(Exception):
     """Exception raised on bad http status"""
 
-    pass
+    def __init__(self, message="", error_status=None):
+        super().__init__(message)
+        self.status = error_status
```
Comment (gtopper) [inline v3io/dataplane/response.py]: How about
```suggestion
    def __init__(self, message, status_code):
        super().__init__(message)
        self.status_code = status_code

    def __repr__(self):
        return f"HttpResponseError('{self}', {self.status_code})"
```

File: v3io/dataplane/response.py [hunk 2]
```diff
@@ -67,7 +70,9 @@ def raise_for_status(self, expected_statuses=None):
         if (expected_statuses is None and self.status_code >= 300) or (
             expected_statuses and self.status_code not in expected_statuses
         ):
-            raise HttpResponseError("Request failed with status {0}: {1}".format(self.status_code, self.body))
+            raise HttpResponseError(
```
Comment (gtopper) [inline v3io/dataplane/response.py:76]: Would be nice to add a test to `tests/test_client_errors.py`.

Comment (tomerm-iguazio) [reply v3io/dataplane/response.py:76]: done

File: tests/test_client_errors.py [hunk 3]
```diff
@@ -107,3 +109,14 @@ def test_error_on_use_after_close():
     client.close()
     with pytest.raises(RuntimeError):
         client.object.get("doesntexist", "doesntexist", raise_for_status=v3io.dataplane.RaiseForStatus.never)
+
+
+@pytest.mark.parametrize("object_functions", [Model.get, Model.put, Model.delete])
```
Comment (gtopper) [inline tests/test_client_errors.py]:
```suggestion
@pytest.mark.parametrize("object_function", [Model.get, Model.put, Model.delete])
```

File: tests/test_client_errors.py [hunk 4]
```diff
@@ -107,3 +109,14 @@ def test_error_on_use_after_close():
...
+def test_raise_http_response_error(object_functions):
+    client = v3io.dataplane.Client(
+        transport_kind="httpclient",
+    )
+    with pytest.raises(HttpResponseError) as response_error:
+        object_functions(client.object, "not-exists", "path/to/object")
+    assert response_error.value.status_code == 404
+    assert "Container not found" in str(response_error.value)
```
Comment (gtopper) [inline tests/test_client_errors.py]:
```suggestion
    with pytest.raises(HttpResponseError, match="Container not found") as response_error:
        object_functions(client.object, "not-exists", "path/to/object")
    assert response_error.value.status_code == 404
```

**Review by gtopper — COMMENTED** [review]
Looks good otherwise 👍

**Review by gtopper — APPROVED** [review]
✅

**Pull request merged**